internal/infrastructure/storage/db/pg/db_manager.go

```diff
@@ -0,0 +1,174 @@
+package postgresdb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+	log "github.com/sirupsen/logrus"
+	"github.com/tdex-network/tdex-daemon/internal/infrastructure/storage/db/pg/sqlc/queries"
+
+	"github.com/tdex-network/tdex-daemon/internal/core/domain"
+	"github.com/tdex-network/tdex-daemon/internal/core/ports"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+const (
+	postgresDriver             = "pgx"
+	insecureDataSourceTemplate = "postgresql://%s:%s@%s:%d/%s?sslmode=disable"
+
+	uniqueViolation = "23505"
+	pgxNoRows       = "no rows in result set"
+)
+
+type repoManager struct {
+	pgxPool *pgxpool.Pool
+	querier *queries.Queries
+
+	marketRepository     domain.MarketRepository
+	tradeRepository      domain.TradeRepository
+	depositRepository    domain.DepositRepository
+	withdrawalRepository domain.WithdrawalRepository
+}
+
+func NewService(dbConfig DbConfig) (ports.RepoManager, error) {
```

Comment:
to make things much simpler, we can expect here 2 strings, one being the pg connect URL in the formal `postgresql://user:password@host:port/name` and the other being the migration source path.

This way we can expose just 2 env vars at config level `TDEX_PG_CONNECT_ADDR` and `TDEX_PG_MIGRATION_SOURCE`. This of course implies validating the pg connect addr, which I think it's fine to be done here in this method.

Reply:
@altafan are u sure about this? in all project we used env var for each db info, changing this also changes out deployment habits a bit

Reply:
Yes, definitely better to use 1 env var rather than 4. In the end, the SDK uses the string URL too instead of 4 distinct params. I'd like to make the same changes in other projects too to simplify the configuration.